cmd/ontap-cvo.go

```diff
@@ -120,6 +120,54 @@ func createS3User(onPremName string, namespaceStr string, client *kubernetes.Cli
 	return true
 }
 
+/*
+This will create the S3 bucket. Requires the bucketName to be hashed, the nasPath and relevant management and svm information
+https://docs.netapp.com/us-en/ontap-restapi/ontap/post-protocols-s3-buckets.html
+*/
+func createS3Bucket(svmInfo svmInfo, mgmInfo managementInfo, bucketName string, nasPath string) bool {
+	hashedName := bucketName + "todo-implement-this"
+	// Create a string that is valid json, as thats the simplest way of working with this request
+	// https://go.dev/play/p/xs_B0l3HsBw
+	jsonString := fmt.Sprintf(
+		`{
+			"comment": "created via the ZONE controller",
+			"name": "%s",
+			"nas_path": "%s",
+			"type": "nas",
+			"policy" : {
+				"statements": [
+					{
+						"effect": "allow",
+						"actions": [
+							"GetObject",
+							"PutObject",
+							"ListBucket",
+							"GetBucketAcl",
+							"GetObjectAcl"
+						],
+						"resources": [
+							"%s",
+							"%s/*"
+						]
+					}
+				]
+			}
+		}`,
+		hashedName, nasPath, hashedName, hashedName)
+	// https://discourse.gohugo.io/t/use-same-argument-twice-in-a-printf-clause/20398
+	urlString := "https://" + mgmInfo.managementIP + "/api/protocols/s3/services/" + svmInfo.svmUUID + "/buckets"
+	statusCode, _ := performHttpPost(mgmInfo.username, mgmInfo.password, urlString, []byte(jsonString))
+	if statusCode == 202 {
```

Comment:
What is the difference between an s3 bucket **job** getting created, and a s3 bucket getting created? Do we need to care about one more than the other?

Reply:
I believe if the API is busy doing something and can't create it right away it will just submit a job to create it, though anecdotally I havent encountered the "job" version when testing Pat's commands. We may want to check for the success first (201) though

Reply:
we may also want to just do a check if S3bucket exists like after 5 seconds im not sure whats expected for the timing

Reply:
Oh okay, that makes sense. Yeah maybe we'll need to have some procedure in place to wait for creation to be done when we receive a job response, We don't want to try to create a new share in that new bucket if we know that it might not be actually done created yet.